src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -308,19 +308,16 @@ class WelcomePage extends Disposable {
 				recent.classList.add('emptyRecent');
 				return;
 			}
-			const ul = container.querySelector('.recent ul');
+			const ul = container.querySelector('.recent ul') as HTMLElement;
 			if (!ul) {
 				return;
 			}
 			const workspacesToShow = workspaces.slice(0, 5);
-			const updateEntries = async () => {
-				while (ul.firstChild) {
-					ul.removeChild(ul.firstChild);
-				}
-				await this.mapListEntries(workspacesToShow, fileService, container);
+			const updateEntries = () => {
+				clearNode(ul);
+				this.mapListEntries(workspacesToShow, fileService, container, ul);
```

Comment:
Purposely not awaiting this call anymore?

Reply:
That was unintentional, fixed.